diagrams/__init__.py

```diff
@@ -296,11 +298,20 @@ def __init__(self, label: str = "", **attrs: Dict):
         self._id = self._rand_id()
         self.label = label
 
+        # Node must be belong to a diagrams.
+        self._diagram = getdiagram()
+        if self._diagram is None:
+            raise EnvironmentError("Global diagrams context not set up")
+        
+        if self._diagram.autolabel:
+            # import pdb; pdb.set_trace()
+            self.label = self.__class__.__name__ + "\n" + self.label
```

Comment:
We should consider a case of `self.label` is an empty string. In this case, the label string will be ended with a new line, and it looks like a bug. I'd like to suggest some solutions.

```python
if self._diagram.autolabel:
    self.label = :"\n".join(filter(None, [self.__class__.__name__, self.label]))

# or

if self._diagram.autolabel:
    prefix = self.__class__.__name__
    if self.label:
        self.label = prefix + "\n" + self.label
    else:
        self.label = prefix

Reply:
makes sense! updated to use the second case since i think it is more readable.